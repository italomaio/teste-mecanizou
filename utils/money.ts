export const formatter = new Intl.NumberFormat("pt-BR", {
  style: "currency",
  currency: "BRL",
  minimumFractionDigits: 2,
});

export function calculateDiscountPrice(
  price: number,
  discountPercentage: number
) {
  const discountFactor = discountPercentage / 100;
  const newPrice = price * (1 - discountFactor);

  return {
    oldPrice: formatter.format(price),
    newPrice: formatter.format(newPrice),
  };
}
