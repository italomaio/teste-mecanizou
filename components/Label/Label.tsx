import { cn } from "@/utils/classes";
import React, { forwardRef } from "react";

type LabelProps = React.LabelHTMLAttributes<HTMLLabelElement>;

const Label = forwardRef<HTMLLabelElement, LabelProps>(
  ({ children, className, ...props }, ref) => {
    return (
      <label
        ref={ref}
        className={cn(
          "inline-block p-0 mb-2 text-left font-medium text-sm leading-none peer-disabled:cursor-not-allowed peer-disabled:opacity-70 text-gray-800 dark:text-white",
          className
        )}
        {...props}
      >
        {children}
      </label>
    );
  }
);

Label.displayName = "Label";

export default Label;
