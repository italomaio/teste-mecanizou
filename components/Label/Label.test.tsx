import { render } from "@/tests/utils";
import { screen } from "@testing-library/react";

import Label from "./Label";
import React from "react";

describe("Label Component", () => {
  it("Should render the label with children", () => {
    render(<Label>Test Label</Label>);
    expect(screen.getByText("Test Label")).toBeInTheDocument();
  });

  it("Should apply custom className", () => {
    render(<Label className="custom-class">Test Label</Label>);
    const labelElement = screen.getByText("Test Label");
    expect(labelElement).toHaveClass("custom-class");
  });

  it("Should forward additional props to the label element", () => {
    render(<Label htmlFor="test-input">Test Label</Label>);
    const labelElement = screen.getByText("Test Label");
    expect(labelElement).toHaveAttribute("for", "test-input");
  });

  it("Should handle ref forwarding", () => {
    const ref = React.createRef<HTMLLabelElement>();
    render(<Label ref={ref}>Test Label</Label>);
    expect(ref.current).toBeInstanceOf(HTMLLabelElement);
  });

  it("Should apply default styles", () => {
    render(<Label>Test Label</Label>);
    const labelElement = screen.getByText("Test Label");
    expect(labelElement).toHaveClass(
      "inline-block p-0 mb-2 text-left font-medium text-sm leading-none peer-disabled:cursor-not-allowed peer-disabled:opacity-70 text-gray-800 dark:text-white"
    );
  });

  it("Should match snapshot", () => {
    const { asFragment } = render(<Label>Test Label</Label>);
    expect(asFragment()).toMatchSnapshot();
  });
});
