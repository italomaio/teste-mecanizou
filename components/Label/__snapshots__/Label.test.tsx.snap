// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`Label Component Should match snapshot 1`] = `
<DocumentFragment>
  <label
    class="inline-block p-0 mb-2 text-left font-medium text-sm leading-none peer-disabled:cursor-not-allowed peer-disabled:opacity-70 text-gray-800 dark:text-white"
  >
    Test Label
  </label>
</DocumentFragment>
`;
